projects/ui-server/ui-client.ts

```diff
@@ -1,95 +1,110 @@
   userID: string;
-  avatar: string;
   soundList: string[];
+
+  constructor(userRes: AxiosResponse) {
+    this.name = userRes.data.username;
+    this.userID = userRes.data.id;
+    this.avatar = userRes.data.avatar;
+  }
 }
 
-export default class UIClient {
-  public accessToken: string;
-  public refreshToken: string;
-  public userData: UserData;
-  private botConfig: AxiosRequestConfig;
+const authURL = `https://discord.com/api/oauth2/authorize?client_id=${ environment.clientID }&redirect_uri=${ encodeURI(environment.UIServerURL) }&response_type=code&scope=identify&prompt=none`;
 
-  constructor(cookies?: any) {
-    this.accessToken = cookies?.accesstoken ?? '';
-    this.refreshToken = cookies?.refreshtoken ?? '';
-    this.userData = {
-      name: '',
-      userID: '',
-      avatar: '',
-      soundList: [],
-    };
-    this.botConfig = { headers: { Authorization: environment.botApiKey } };
-  }
+function getTokensFromCode(authCode: string) {
+  const params = new URLSearchParams({
+    client_id: environment.clientID,
+    client_secret: environment.clientSecret,
+    grant_type: 'authorization_code',
+    code: authCode,
+    redirect_uri: environment.UIServerURL,
+  });
+  return axios.post('https://discord.com/api/oauth2/token', params);
+}
 
-  async authenticate(authCode?: string) {
-    const params = new URLSearchParams({
-      client_id: environment.clientID,
-      client_secret: environment.clientSecret,
-    });
-    if (authCode) {
-      params.append('grant_type', 'authorization_code');
-      params.append('code', authCode);
-      params.append('redirect_uri', environment.UIServerURL);
-    } else {
-      params.append('grant_type', 'refresh_token');
-      params.append('refresh_token', this.refreshToken);
-    }
+function getTokensFromRefresh(refreshToken: string) {
+  const params = new URLSearchParams({
+    client_id: environment.clientID,
+    client_secret: environment.clientSecret,
+    grant_type: 'refresh_token',
+    refresh_token: refreshToken,
+  });
+  return axios.post('https://discord.com/api/oauth2/token', params);
+}
+
+export const discordAuth: RequestHandler = async (req, res, next) => {
+  if (req.query.code) {
     try {
-      const res = await axios.post('https://discord.com/api/oauth2/token', params);
-      this.accessToken = res.data.access_token;
-      this.refreshToken = res.data.refresh_token;
+      const tokenRes = await getTokensFromCode(String(req.query.code));
+      res.cookie('accesstoken', tokenRes.data.access_token, { httpOnly: true, maxAge: 1000 * 60 * 30 });
+      res.cookie('refreshtoken', tokenRes.data.refresh_token, { httpOnly: true, maxAge: 1000 * 60 * 60 * 48 });
       console.log('token response from discord');
     } catch (error) {
       console.log(error);
     }
+    res.redirect('/');
+    return;
   }
 
-  async getUser(retry?: boolean) {
+  let userReqToken: string;
+  if (req.cookies.accesstoken) userReqToken = req.cookies.accesstoken;
```

Comment:
```suggestion
  const userReqToken = req.cookies.accesstoken;
```

The `if` doesn't look like it really does anything.

Reply:
Yeah it'll need to be `let` but that's fine.

I guess cookies is there but just empty, I was thinking that it would explode when a new user hit that line and you'd have to at least `req.cookies?.accesstoken`

Reply:
The assignment won't blow up unless `cookies` is null, it doesn't care if `accesstoken` is null, it'll happily assign it anyway.